src/plugins/unified_search/public/query_string_input/query_bar_top_row.tsx

```diff
@@ -308,85 +337,171 @@ export const QueryBarTopRow = React.memo(
             dateFormat={uiSettings.get('dateFormat')}
             isAutoRefreshOnly={showAutoRefreshOnly}
             className="kbnQueryBar__datePicker"
+            isQuickSelectOnly={isMobile ? false : isQueryInputFocused}
+            width={isMobile ? 'full' : 'auto'}
+            compressed={shouldShowDatePickerAsBadge()}
           />
         </EuiFlexItem>
       );
     }
 
     function renderUpdateButton() {
+      if (!shouldRenderUpdatebutton()) {
+        return null;
+      }
+
+      const buttonLabelUpdate = i18n.translate('unifiedSearch.queryBarTopRow.submitButton.update', {
+        defaultMessage: 'Needs updating',
+      });
+      const buttonLabelRefresh = i18n.translate(
+        'unifiedSearch.queryBarTopRow.submitButton.refresh',
+        {
+          defaultMessage: 'Refresh query',
+        }
+      );
+
       const button = props.customSubmitButton ? (
         React.cloneElement(props.customSubmitButton, { onClick: onClickSubmitButton })
       ) : (
-        <SuperUpdateButton
-          needsUpdate={props.isDirty}
-          isDisabled={isDateRangeInvalid}
-          isLoading={props.isLoading}
-          onClick={onClickSubmitButton}
-          fill={props.fillSubmitButton}
-          data-test-subj="querySubmitButton"
-        />
+        <EuiFlexItem grow={false}>
+          <SuperUpdateButton
+            iconType={props.isDirty ? 'kqlFunction' : 'refresh'}
+            iconOnly
+            aria-label={props.isLoading ? buttonLabelUpdate : buttonLabelRefresh}
+            isDisabled={isDateRangeInvalid}
+            isLoading={props.isLoading}
+            onClick={onClickSubmitButton}
+            size={shouldShowDatePickerAsBadge() ? 's' : 'm'}
+            color={props.isDirty ? 'success' : 'primary'}
+            fill={props.isDirty}
+            data-test-subj="querySubmitButton"
+            // @ts-expect-error Need to fix expecting `children` in EUI
+            toolTipProps={{
```

Comment:
nit: seems like all other props have stable reference and could have prevented re-render of the SuperUpdateButton. If not create stable reference then better remove React.memo wrapper since it is not working anymore